File: backend/src/main/java/turip/region/controller/dto/response/CountryResponse.java
package turip.region.controller.dto.response;

import com.fasterxml.jackson.annotation.JsonProperty;
import turip.region.domain.Country;

public record CountryResponse(
        Long id,
        @JsonProperty("countryName")
        String name,
        @JsonProperty("countryImageUrl")
        String imageUrl
) {

    public static CountryResponse from(Country country) {
        return new CountryResponse(
                country.getId(),
                country.getName(),
                country.getImageUrl()
        );
    }
}

File: android/app/src/main/java/com/on/turip/data/region/dto/CountryResponse.kt
package com.on.turip.data.region.dto

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

@Serializable
data class CountryResponse(
    @SerialName("countryImageUrl")
    val countryImageUrl: String,
    @SerialName("countryName")
    val countryName: String,
    @SerialName("id")
    val id: Int,
)
